### ISSUE TITLE
new lecture and CAs uploaded

### ISSUE THREAD

**Issue by hshokrig** [hshokrig/EP3260-MLoNs-2023#7]
You have the last lecture slides and CAs 6 and 7, last ones. The deadline for CAs is Apr 15.

**Comment by hshokrig** [comment]
Our MLoN workshop is on March 30 and 31. 

Each Study Group (SG) is supposed to prepare a 45 minutes oral presentation, followed by 15 minutes Q&A. Moreover, each SG should design a poster from their presentation for the poster session to take additional questions we cannot take in Q&A. .

In the following, I provided some potential topics. But you are welcome to suggest new topics as well. You should choose some prominent papers (does not have to be new or the ones we suggested) to cover various aspects of that topic. Send me ASAP, by replying to this issue, a list of three topics for your group (with your priority ranks). We will assign the topics to you afterward and your scheduled presentation time. ALL MEMBERS SHOULD PRESENT ON THE VIRTUAL WORKSHOP DAYS IF THEY WANT THE COURSE CREDIT. Exceptions are allowed for emergency reasons but you should inform us a priory if you cannot attend.

**Special Topic 1: Privacy and security in distributed learning**

- Kairouz, Peter, et al. "Advances and open problems in federated learning." arXiv preprint arXiv:1912.04977 (2019).
- Abadi, Martin, et al. "Deep learning with differential privacy." Proceedings of the 2016 ACM SIGSAC Conference on Computer and Communications Security. 2016.
- Dwork, Cynthia, and Aaron Roth. "The algorithmic foundations of differential privacy." Foundations and Trends® in Theoretical Computer Science 9.3–4 (2014): 211-407.
- Du, Wenliang, and Mikhail J. Atallah. "Secure multi-party computation problems and their applications: a review and open problems." Proceedings of the 2001 workshop on New security paradigms. 2001.

**Special Topic 2: Handling Non-IID datasets**

- Zhao, Yue, et al. "Federated learning with non-iid data." arXiv preprint arXiv:1806.00582 (2018).
- Sattler, Felix, et al. "Robust and communication-efficient federated learning from non-iid data." IEEE transactions on neural networks and learning systems (2019).
- Li, Xiang, et al. "On the convergence of fedavg on non-iid data." arXiv preprint arXiv:1907.02189 (2019).

**Special Topic 3: Model Compression**

- Cheng, Yu, et al. "A survey of model compression and acceleration for deep neural networks." arXiv preprint arXiv:1710.09282 (2017).
- Han, Song, Huizi Mao, and William J. Dally. "Deep compression: Compressing deep neural networks with pruning, trained quantization and huffman coding." arXiv preprint arXiv:1510.00149 (2015).
- Iandola, Forrest N., et al. "SqueezeNet: AlexNet-level accuracy with 50x fewer parameters and< 0.5 MB model size." arXiv preprint arXiv:1602.07360 (2016).

**Special Topic 4: Error compensation and variance reduction**

- Stich, Sebastian U., and Sai Praneeth Karimireddy. "The error-feedback framework: Better rates for SGD with delayed gradients and compressed communication." arXiv preprint arXiv:1909.05350 (2019).
- Horváth, Samuel, et al. "Stochastic distributed learning with gradient quantization and variance reduction." arXiv preprint arXiv:1904.05115 (2019).

**Comment by vnmo** [comment]
Hi Hossein,

Regarding the workshop topic I was thinking about a topic that we are currently working on extensively. We have recently submitted a paper in this topic. Is it allowed to present a topic that we are working on?

The major related works of the topic comes from DNN partitioning and we are trying to achieve some similar goals in an IoT/EdgeComputing scenario in a very different mechanism. A quick abstract of one of the work is given below for your understanding.

If allowed, I would like to take this topic, with a title “Hierarchical Inference to reduce Client-Server communication in ML networks”.

We consider a resource-constrained Edge Device (ED), such as an IoT sensor or a microcontroller unit, embedded with a small-size ML model (S-ML) for a generic classification application, and an Edge Server (ES) that hosts a large-size ML model (L-ML). Since the inference accuracy of S-ML is lower than that of the L-ML, offloading all the data samples to the ES results in high inference accuracy, but it defeats the purpose of embedding S-ML on the ED and deprives the benefits of reduced latency, bandwidth savings, and energy efficiency of doing local inference. In order to get the best out of both worlds, i.e., the benefits of doing inference on the ED and the benefits of doing inference on ES, we explore the idea of Hierarchical Inference (HI), wherein S-ML inference is only accepted when it is correct, otherwise the data sample is offloaded for L-ML inference. However, the ideal implementation of HI is infeasible as the correctness of the S-ML inference is not known to the ED. We thus propose an online meta-learning framework that the ED can use to predict the correctness of the S-ML inference. In particular, we propose to use the probability corresponding to the maximum probability class output by S-ML for a data sample and decide whether to offload it or not. The resulting online learning problem turns out to be a Prediction with Expert Advice (PEA) problem with continuous expert space. We consider two scenarios, a full feedback scenario, where the ED receives feedback on the correctness of the S-ML once it accepts the inference, and a no-local feedback scenario. We propose the HIL-F and HIL-N algorithms and prove that both of them has sublinear regret bounds without any assumption on the smoothness of the loss function. We evaluate and benchmark the performance of the proposed algorithms for image classification application using different datasets.


Regards,
Vishnu

On 12 Mar 2023, at 18:02, Hana ***@***.***> wrote:



Assigned #7<https://github.com/hshokrig/EP3260-MLoNs-2023/issues/7> to @vnmo<https://github.com/vnmo>.

**Comment by ozanalptopal** [comment]
Hi!


We are group 1 (Yasaman, Firooz and Ozan). We are mainly interested in 4th one about distributed variance reduction and decentralized constrained optimization since Firooz has been working in this area. We have a list of papers that we are planning to present :

  1.  Double Averaging and Gradient Projection: Convergence Guarantees for Decentralized Constrained Optimization( https://arxiv.org/abs/2210.03232 )
  2.  Push–Pull Gradient Methods for Distributed Optimization in Networks (https://arxiv.org/abs/2210.03232 )
  3.
Distributed Subgradient Projection Algorithm Over Directed Graphs (https://ieeexplore.ieee.org/document/7582416)


In this case, our ranking is:


4th topic

2nd topic

3rd topic



Best regards,
Ozan

________________________________

**Comment by lonaparte** [comment]
Hi all!

We are group 3 (Jeannie, Li, Yifei and Yusen). We’d like to work on topic 1 as it is closely related to the research projects that some of us are working on.  

Here is our ranking: topic 1, 3, 2. 

Best regards,
Li

**Comment by GaneshSeeram** [comment]
Hi,

We are from group 2 (Zinat, Hansi, Eren, Irshad) and our preference is as follows
Topic 2 , Topic 4, and Topic 3.

Regards,
Satya.

**Comment by hshokrig** [comment]
group 1, @ozanalptopal : topic 4

group 2, @lonaparte : topic 1

group 3, @GaneshSeeram : topic 2

**Comment by ozanalptopal** [comment]
Dear Professor Shokri-Ghadikolaei,


I am writing this email to become sure that our group has understood
everything correctly regarding presentations at the FMLoN workshop.

We are assigned the fourth topic, which is related to error compensation
and variance reduction in distributed optimization algorithms.


As you mentioned in your previous email, we are allowed to choose other
related papers instead of the ones that you suggested.

We suggested three papers. The first paper is my own work and I am very
interested in presenting it in the workshop and getting some feedback.

The algorithm removes the variance caused by heterogeneous local objective
functions and handles local constraints for the first time, so it is
related.


*The allocation of the fourth topic to our group means that we can present
this paper <https://arxiv.org/abs/2210.03232>, isn't it?*


Best regards,

Group one
